runtime/jcl/common/java_dyn_methodhandle.c

```diff
@@ -95,9 +96,22 @@ lookupInterfaceMethod(J9VMThread *currentThread, J9Class *lookupClass, J9UTF8 *n
 			vmFuncs->setCurrentExceptionNLS(currentThread, J9VMCONSTANTPOOL_JAVALANGINCOMPATIBLECLASSCHANGEERROR, J9NLS_JCL_PRIVATE_INTERFACE_REQUIRES_INVOKESPECIAL);
 			method = NULL;
 		} else {
-			*methodIndex = getITableIndexForMethod(method, lookupClass);
-			if (*methodIndex == -1) {
-				method = NULL;
+			if (J9_ARE_ANY_BITS_SET(J9_CLASS_FROM_METHOD(method)->romClass->modifiers, J9_JAVA_INTERFACE)) {
+				*methodIndex = getITableIndexForMethod(method, lookupClass);
+				if (-1  == *methodIndex) {
+					char msg[64]; /* long enough for most name and signature combinations */
```

Comment:
Let's write the code to allocate a buffer if this assumption is invalid.  Not uncommon for ~40+ chars to be consumed by the package name.  We've got examples of that pattern in exceptionsupport.c (I think)